tests/test_cli_compile.py

```diff
@@ -3771,3 +3772,70 @@ def test_stdout_should_not_be_read_when_stdin_is_not_a_plain_file(
     out = runner.invoke(cli, [req_in.as_posix(), "--output-file", fifo.as_posix()])
 
     assert out.exit_code == 0, out
+
+
+def test_compile_inline_script_metadata(runner, tmp_path, current_resolver):
+    (tmp_path / "script.py").write_text(
+        dedent(
+            """
+            # /// script
+            # dependencies = [
```

Comment:
Should we also have tests for extras?

Reply:
Oh, there's no explicit extras in the spec. But how about `requires-python`?

Reply:
I'm not sure how that would influence the build process?

Reply:
It should reject incompatible runtimes, for example.